### ISSUE TITLE
Automated versioning

### ISSUE THREAD

**Issue by EmperorYP7** [casbin/casbin-cpp#171]
### Description

The [version specified in CMake config](https://github.com/casbin/casbin-cpp/blob/f9d1a3394b075361f32a00a24dd10710643e5967/CMakeLists.txt#L36) and the [version released by GitHub Actions](https://github.com/casbin/casbin-cpp/releases) is not the same and can be really confusing for the clients. We can automate the versioning to coincide the CMake version with the GH version of casbin-cpp through another GitHub Action which should trigger on push to `master` branch.

**Comment by hsluoyz** [comment]
@sheny1xuan

**Comment by sheny1xuan** [comment]
Good point. But I think it's maybe difficult to do that automatically. Because if we update cmake file atfer a new release, we will generate change in `cmakelists.txt`. And I can't find a plugin can do that thing. Maybe we can update cmake version with the [release rule](https://github.com/semantic-release/commit-analyzer). This is a very tricky thing.

**Comment by EmperorYP7** [comment]
I'll keep everyone updated once I have a practical approach for this.
